enkit/machinecert/commands.go

```diff
@@ -0,0 +1,89 @@
+// Package machinecert provides the machine-cert subcommands to enkit.
+package machinecert
+
+import (
+	"fmt"
+
+	"github.com/enfabrica/enkit/lib/client"
+
+	"github.com/spf13/cobra"
+)
+
+type Root struct {
+	*cobra.Command
+	*client.BaseFlags
+}
+
+func New(base *client.BaseFlags) (*Root, error) {
+	root, err := NewRoot(base)
+	if err != nil {
+		return nil, err
+	}
+
+	root.AddCommand(NewPrint(root).Command)
+	root.AddCommand(NewInstall(root).Command)
+
+	return root, nil
+}
+
+func NewRoot(base *client.BaseFlags) (*Root, error) {
+	rc := &Root{
+		Command: &cobra.Command{
+			Use:   "machine-cert",
+			Short: "Commands for managing a machine's SSH certificate",
+			Long:  `machine-cert - commands for managing a machine's SSH certificate`,
+		},
+		BaseFlags: base,
+	}
+
+	// TODO: add root-level flags
+
+	return rc, nil
+}
+
+type Print struct {
+	*cobra.Command
+	root *Root
+}
+
+func NewPrint(root *Root) *Print {
+	command := &Print{
+		Command: &cobra.Command{
+			Use:     "print",
```

Comment:
print ... would... read a new certificate from ...? I was expecting a "write" or "generate" that generates a certificate and writes it into a file. Maybe a sign to take an existing certificate and have it signed by the server. Maybe an install like the one you have to setup ssh (or authorized_keys file) as necessary.

Reply:
As we discussed in chat, in the interest of time - going to move ahead with a single command that does the whole flow, with flags to disable disruptive portions/short-circuit portions, as this is more likely to be useful in the short term.